src/protocol.rs

```diff
@@ -394,8 +398,46 @@ impl std::fmt::Display for ParticipantIdentifier {
     }
 }
 
-/// A generic identifier
 #[derive(Copy, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+
+/// An [`Identifier`] is a session identifier that uniquely identifies a single
+/// instance of a protocol and all messages associated with it.
+///
+/// [`Identifier`] are globally unique identifiers that allow parties to
+/// distinguish between different sessions. The "globally unique" property
+/// of these identifiers forces parties to avoid any collisions between
+/// different sessions and any kind of replay attack by associating messages,
+/// parameters, proofs and commitments with their corresponding sessions.
+///
+/// Discrepancies with the paper with respect to session identifiers:
```

Comment:
suggestion: make this a separate section with `#` (e.g. [like here](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#documenting-components))